java/src/main/java/org/apache/arrow/gandiva/evaluator/Filter.java

```diff
@@ -0,0 +1,145 @@
+ * Follow these steps to use this class:
+ * 1) Use the static method make() to create an instance of this class that evaluates a
+ * condition.
+ * 2) Invoke the method evaluate() to evaluate the filter against a RecordBatch
+ * 3) Invoke close() to release resources
+ */
+public class Filter {
+  private static final org.slf4j.Logger logger = org.slf4j.LoggerFactory.getLogger(Filter.class);
+
+  private final long moduleId;
+  private final Schema schema;
+  private boolean closed;
+
+  private Filter(long moduleId, Schema schema) {
+    this.moduleId = moduleId;
+    this.schema = schema;
+    this.closed = false;
+  }
+
+  /**
+   * Invoke this function to generate LLVM code to evaluate the condition expression.
+   * Invoke Filter::Evaluate() against a RecordBatch to evaluate the filter on
+   * this record batch
+   *
+   * @param schema    Table schema. The field names in the schema should match the fields used
+   *                  to create the TreeNodes
+   * @param condition condition to be evaluated against data
+   * @return A native filter object that can be used to invoke on a RecordBatch
+   */
+  public static Filter make(Schema schema, Condition condition) throws GandivaException {
+    return make(schema, condition, ConfigurationBuilder.getDefaultConfiguration());
+  }
+
+  /**
+   * Invoke this function to generate LLVM code to evaluate the condition expression.
+   * Invoke Filter::Evaluate() against a RecordBatch to evaluate the filter on
+   * this record batch
+   *
+   * @param schema          Table schema. The field names in the schema should match the fields used
+   *                        to create the TreeNodes
+   * @param condition       condition to be evaluated against data
+   * @param configurationId Custom configuration created through config builder.
+   * @return A native evaluator object that can be used to invoke these projections on a RecordBatch
+   */
+  public static Filter make(Schema schema, Condition condition, long configurationId)
+      throws GandivaException {
+    // Invoke the JNI layer to create the LLVM module representing the filter.
+    GandivaTypes.Condition conditionBuf = condition.toProtobuf();
+    GandivaTypes.Schema schemaBuf = ArrowTypeHelper.arrowSchemaToProtobuf(schema);
+    JniWrapper gandivaBridge = JniWrapper.getInstance();
+    long moduleId = gandivaBridge.buildFilter(schemaBuf.toByteArray(),
+        conditionBuf.toByteArray(), configurationId);
+    return new Filter(moduleId, schema);
+  }
+
+  /**
+   * Invoke this function to evaluate a set of expressions against a recordBatch.
+   *
+   * @param recordBatch Record batch including the data
+   * @param selectionVector  Result of applying the filter on the data
+   */
+  public void evaluate(ArrowRecordBatch recordBatch, SelectionVector selectionVector)
+      throws GandivaException {
+
+    if (this.closed) {
+      throw new EvaluatorClosedException();
+    }
+
+    int numRows = recordBatch.getLength();
```

Comment:
should we also cross validate the record batch schema with make schema here to fail fast..this is not present in projector too..maybe add in both?
(looks like we do that in projector and filter in gandiva core..)

Reply:
evaluate codepath is very performance sensitive. can't afford to do the same check twice.